src/com/scottyplunkett/server/FormContentResponse.java

```diff
@@ -0,0 +1,41 @@
+package com.scottyplunkett.server;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.scottyplunkett.server.ByteArraysReducer.merge;
+
+public class FormContentResponse {
+    private final Path formPath = Paths.get("pages/form.html");
+    private HTTPRequest httpRequest;
+    private String method;
+    private byte[] head;
+    private byte[] body;
+    private byte[] responseContent;
+
+
+    FormContentResponse(HTTPRequest request, String date) throws IOException {
+        httpRequest = request;
+        method = Parser.findRequestMethod(httpRequest.getRequestLine());
+        if(method.equals("POST") || method.equals("PUT")) writeToForm();
```

Comment:
These conditionals (side-effects) could be extracted to private methods.

Reply:
Agreed, they will be in an upcoming PR. Thanks.